r1cs/gadgets/std/src/fields/nonnative/nonnative_field_gadget.rs

```diff
@@ -1985,3 +1919,101 @@ impl<SimulationF: PrimeField, ConstraintF: PrimeField> EqGadget<ConstraintF>
         Ok(())
     }
 }
+
+/// function employed in tests to evaluate the difference in terms of constraints between
+/// performing reduction after each multiplication and performing a reduction only when
+/// necessary, i.e., after several muls. Currently applicable only to pseudo-mersenne fields,
+/// as the product in a non pseudo-mersenne field cannot be multiplied to other field elements
+/// without being reduced
+#[cfg(test)]
+pub(crate) fn bench_mul_without_reduce<SimulationF: PrimeField, ConstraintF: PrimeField,R: rand::RngCore>(rng: &mut R, surfeit: usize) {
```

Comment:
Let's remove this from the PR and maybe save it somewhere for later

Reply:
I can mark the test with #[ignore] to avoid running it unless explicitly requested, as it seems to me the best place to save this bench for later reuse. What do you think?